Dismount on 1.14 on 1.8 server

**Describe the bug, provide any errors**
A clear and concise description of what the bug is. Can you https://hastebin.com the error?
hastebin: https://hastebin.com/raw/olabixovop

-------If hastebin doesn't work, use this error--------------------------------------------

30.08 21:34:26 [Server] INFO at java.lang.Thread.run(Unknown Source) [?:1.8.0_201]
30.08 21:34:26 [Server] INFO Caused by: java.lang.NullPointerException
30.08 21:34:26 [Server] INFO at us.myles.ViaVersion.bukkit.listeners.multiversion.PlayerSneakListener.setHeight(PlayerSneakListener.java:125) ~[ViaVersion-2.1.3.jar:?]
30.08 21:34:26 [Server] INFO at us.myles.ViaVersion.bukkit.listeners.multiversion.PlayerSneakListener.playerToggleSneak(PlayerSneakListener.java:76) ~[ViaVersion-2.1.3.jar:?]
30.08 21:34:26 [Server] INFO at sun.reflect.GeneratedMethodAccessor255.invoke(Unknown Source) ~[?:?]
30.08 21:34:26 [Server] INFO at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_201]
30.08 21:34:26 [Server] INFO at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_201]
30.08 21:34:26 [Server] INFO at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:301) ~[spigot-1.8.jar:git-Spigot-29dbaa7-262c777]
30.08 21:34:26 [Server] INFO ... 14 more
----------------------------------------------------------------------------------------------

**How can we reproduce it?**
Steps to reproduce the behavior:
(the server is 1.8)
1. Login on 1.14.4.
2. Ride on boat, horse, or pig.
3. Move around little bit.
4. The rideable entity&player will be teleported to about where I was 10 seconds before.
5. If the rideable entity is moved in server-side or other things (water streams, pistons, etc.) it will be dismounted correctly.
6. This sometimes work, and sometimes not, sometimes, rideable entity can disappear.
7. Use pigs for best results.

**Expected behaviour**
Dismounting without any teleportations or disappearance.

**Screenshots**
If applicable, add screenshots to help explain your problem.
Screenshots doesn't seem enough, so there is a video for it.
[2019-08-3110-44-54.zip](https://github.com/ViaVersion/ViaVersion/files/3561995/2019-08-3110-44-54.zip)



**ViaVersion Dump:**
- Type /viaversion dump, and put the link here.
https://dump.viaversion.com/d4ceba7ceea13b35d20e0a055bdec3e71c2b2178a0fafc1c4fc6d8cfaf3ccd69

**Additional server info**
Do you use a proxy (eg. BungeeCord)? What software do you use and what plugins?
Proxy: no
Plugins: too many, screenshot
![2019-08-31_11 30 34](https://user-images.githubusercontent.com/45235204/64058203-cc2c1800-cbe2-11e9-91f2-74d3d7cc848c.png)

--- a/bukkit/src/main/java/us/myles/ViaVersion/bukkit/listeners/multiversion/PlayerSneakListener.java
+++ b/bukkit/src/main/java/us/myles/ViaVersion/bukkit/listeners/multiversion/PlayerSneakListener.java
@@ -39,13 +39,13 @@ public class PlayerSneakListener extends ViaBukkitListener {
         final String packageName = plugin.getServer().getClass().getPackage().getName();
         getHandle = Class.forName(packageName + ".entity.CraftPlayer").getMethod("getHandle");
 
+        final Class<?> entityPlayerClass = Class.forName(packageName
+                .replace("org.bukkit.craftbukkit", "net.minecraft.server") + ".EntityPlayer");
         try {
-            setSize = Class.forName(packageName
-                    .replace("org.bukkit.craftbukkit", "net.minecraft.server") + ".EntityPlayer").getMethod("setSize", Float.TYPE, Float.TYPE);
-        } catch (ClassNotFoundException | NoSuchMethodException e) {
+            setSize = entityPlayerClass.getMethod("setSize", Float.TYPE, Float.TYPE);
+        } catch (NoSuchMethodException e) {
             // Don't catch this one
-            setSize = Class.forName(packageName
-                    .replace("org.bukkit.craftbukkit", "net.minecraft.server") + ".EntityPlayer").getMethod("a", Float.TYPE, Float.TYPE);
+            setSize = entityPlayerClass.getMethod("a", Float.TYPE, Float.TYPE);
         }
 
 

--- a/bukkit/src/main/java/us/myles/ViaVersion/bukkit/listeners/multiversion/PlayerSneakListener.java
+++ b/bukkit/src/main/java/us/myles/ViaVersion/bukkit/listeners/multiversion/PlayerSneakListener.java
@@ -31,17 +31,23 @@ public class PlayerSneakListener extends ViaBukkitListener {
     private boolean is1_14Fix;
     private boolean useCache;
 
-    public PlayerSneakListener(ViaVersionPlugin plugin, boolean is1_9Fix, boolean is1_14Fix) {
+    public PlayerSneakListener(ViaVersionPlugin plugin, boolean is1_9Fix, boolean is1_14Fix) throws ReflectiveOperationException {
         super(plugin, null);
         this.is1_9Fix = is1_9Fix;
         this.is1_14Fix = is1_14Fix;
+
+        final String packageName = plugin.getServer().getClass().getPackage().getName();
+        getHandle = Class.forName(packageName + ".entity.CraftPlayer").getMethod("getHandle");
+
         try {
-            getHandle = Class.forName(plugin.getServer().getClass().getPackage().getName() + ".entity.CraftPlayer").getMethod("getHandle");
-            setSize = Class.forName(plugin.getServer().getClass().getPackage().getName()
+            setSize = Class.forName(packageName
                     .replace("org.bukkit.craftbukkit", "net.minecraft.server") + ".EntityPlayer").getMethod("setSize", Float.TYPE, Float.TYPE);
         } catch (ClassNotFoundException | NoSuchMethodException e) {
-            e.printStackTrace();
+            // Don't catch this one
+            setSize = Class.forName(packageName
+                    .replace("org.bukkit.craftbukkit", "net.minecraft.server") + ".EntityPlayer").getMethod("a", Float.TYPE, Float.TYPE);
         }
+
 
         // From 1.9 upwards the server hitbox is set in every entity tick, so we have to reset it everytime
         if (ProtocolRegistry.SERVER_PROTOCOL >= ProtocolVersion.v1_9.getId()) {

--- a/bukkit/src/main/java/us/myles/ViaVersion/bukkit/platform/BukkitViaLoader.java
+++ b/bukkit/src/main/java/us/myles/ViaVersion/bukkit/platform/BukkitViaLoader.java
@@ -82,7 +82,12 @@ public class BukkitViaLoader implements ViaPlatformLoader {
         if (ProtocolRegistry.SERVER_PROTOCOL < ProtocolVersion.v1_14.getId()) {
             boolean use1_9Fix = plugin.getConf().is1_9HitboxFix() && ProtocolRegistry.SERVER_PROTOCOL < ProtocolVersion.v1_9.getId();
             if (use1_9Fix || plugin.getConf().is1_14HitboxFix()) {
-                storeListener(new PlayerSneakListener(plugin, use1_9Fix, plugin.getConf().is1_14HitboxFix())).register();
+                try {
+                    storeListener(new PlayerSneakListener(plugin, use1_9Fix, plugin.getConf().is1_14HitboxFix())).register();
+                } catch (ReflectiveOperationException e) {
+                    Via.getPlatform().getLogger().warning("Could not load hitbox fix - please report this on our GitHub");
+                    e.printStackTrace();
+                }
             }
         }
 

